rbac/ledger_sync/inbound/listener.py

```diff
@@ -48,6 +49,7 @@ def process(rec, conn):
         batch_list = batch_to_list(batch=batch)
         status = ClientSync().send_batches_get_status(batch_list=batch_list)
         if status[0]["status"] == "COMMITTED":
+            rec["data"]["metadata"] = json.loads(rec["data"]["metadata"])
```

Comment:
This assumes that rec["data"]["metadata"] will be in a formatted string where json.loads() can decode it and turn it back into a dictionary. This is not guaranteed to always be true and will throw a JSONDecodeError() when it tries to run json.loads()

Reply:
Yes, this should not be a problem because all the Inbound & Delta Sync data from ADs, and, also the test data loaded will be included with metadata field, always. Please refer the above considerations, again.

**Consideration 1: There are 3 different types of data that sit in the Rethink DB.**

1. Data that come from AD (LDAP & Azure)
2. Test data that is being loaded initially when the project is up.
3. Data that come from NEXT (Server APIs & Client UI)
So for the scenarios 1 & 2, we are considering that data as INBOUND. And, for the scenario 3, we are that data as OUTBOUND.

Reply:
Sounds good for inbound as wet set the metadata field in rbac_transactions.py, still not sure about outbound because of the APIs could also be passed in additional information. Resolving this portion.